packages/react-dom/src/client/ReactDOMFiberInput.js

```diff
@@ -203,6 +209,16 @@ export function postMountWrapper(
   const node = ((element: any): InputWithWrapperState);
 
   if (props.hasOwnProperty('value') || props.hasOwnProperty('defaultValue')) {
+    // Submit or reset inputs should not have their value overridden
+    // with a null or undefined value, or else this will result in a button with
+    // no text, ignoring the browser-default "Submit", etc.
```

Comment:
Can we just shorten this and add a link to https://github.com/facebook/react/issues/12872 here for reference?

> Avoid setting value attribute on submit/reset inputs as it overrides the default value provided by the browser. See: [https://github.com/facebook/react/issues/12872](https://github.com/facebook/react/issues/12872)